data/transactions/logic/assembler.go

```diff
@@ -2125,6 +2125,25 @@ func (ops *OpStream) ReportProblems(fname string) {
 	}
 }
 
+// ReportProblemsString issues accumulated warnings and outputs an error string.
```

Comment:
Is this repeating the same work as in `goal`? Can you call this there?

Reply:
fmt.Fprintf takes an io.Writer, so you should be able to refactor ReportProblems to take an io.Writer too, so that goal calls it with stderr and rest calls with a string.Builder.

Reply:
Take a "prefix" argument too, so you can get the fname in the goal output.

Reply:
Good point -- Should `fname` be left blank for the API call?

Reply:
Yeah, I think so.